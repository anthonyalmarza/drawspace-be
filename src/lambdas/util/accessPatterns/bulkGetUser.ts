import table from 'lambdas/util/table'
import userEntity from 'lambdas/util/entities/user'

import { User } from 'types'

export default async (userIds: string[]): Promise<User[]> => {
    const { Responses: results } = await table.batchGet(
        userIds.map((userId) => userEntity.getBatch({ id: userId }))
    )
    return results[table.name]
}
