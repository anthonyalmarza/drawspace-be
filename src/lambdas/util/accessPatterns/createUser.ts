import userEntity from 'lambdas/util/entities/user'
import { User, UpdateReturnValues } from 'types'

export default async ({
    userId,
    name,
}: {
    userId: string
    name: string
}): Promise<User> => {
    const { Attributes: user } = await userEntity.update(
        { id: userId, name },
        {
            returnValues: UpdateReturnValues.ALL_NEW,
            conditions: { attr: 'pk', exists: false },
        }
    )
    return user
}
