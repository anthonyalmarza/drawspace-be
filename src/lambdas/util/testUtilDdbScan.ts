import DynamoDB from 'aws-sdk/clients/dynamodb'

const { apiTableName } = require('lambdas/util/envVarsRestApi')
/* eslint-enable */
const DocumentClient = new DynamoDB.DocumentClient({
    endpoint: 'http://localhost:9001',
    accessKeyId: 'dynamo',
    secretAccessKey: 'devDummyKey',
    region: 'local',
})

export default () => DocumentClient.scan({ TableName: apiTableName }).promise()
