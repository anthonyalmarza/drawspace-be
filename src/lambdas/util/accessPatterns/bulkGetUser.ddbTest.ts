import { advanceTo } from 'jest-date-mock'
import testUtilDdbScan from 'lambdas/util/testUtilDdbScan'
import createUser from 'lambdas/util/accessPatterns/createUser'
import bulkGetUser from 'lambdas/util/accessPatterns/bulkGetUser'

advanceTo(new Date(2020, 5, 29, 0, 0, 0))

test('bulkGetUser', async () => {
    await Promise.all(
        Array.from({ length: 5 }, (x, i) =>
            createUser({ userId: i.toString(), name: 'mock user name' })
        )
    )

    const res = await bulkGetUser(['0', '1', '2', '3', '4', '5'])
    const scan = await testUtilDdbScan()
    expect(res).toMatchSnapshot()
    expect(scan).toMatchSnapshot()
})
