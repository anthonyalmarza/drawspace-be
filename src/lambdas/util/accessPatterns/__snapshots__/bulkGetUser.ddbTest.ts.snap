// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bulkGetUser 1`] = `
Array [
  Object {
    "created": "2020-06-29T00:00:00.000Z",
    "entity": "user",
    "id": "3",
    "modified": "2020-06-29T00:00:00.000Z",
    "name": "mock user name",
    "privateDrawingCount": 0,
    "publicDrawingCount": 0,
  },
  Object {
    "created": "2020-06-29T00:00:00.000Z",
    "entity": "user",
    "id": "0",
    "modified": "2020-06-29T00:00:00.000Z",
    "name": "mock user name",
    "privateDrawingCount": 0,
    "publicDrawingCount": 0,
  },
  Object {
    "created": "2020-06-29T00:00:00.000Z",
    "entity": "user",
    "id": "2",
    "modified": "2020-06-29T00:00:00.000Z",
    "name": "mock user name",
    "privateDrawingCount": 0,
    "publicDrawingCount": 0,
  },
  Object {
    "created": "2020-06-29T00:00:00.000Z",
    "entity": "user",
    "id": "1",
    "modified": "2020-06-29T00:00:00.000Z",
    "name": "mock user name",
    "privateDrawingCount": 0,
    "publicDrawingCount": 0,
  },
  Object {
    "created": "2020-06-29T00:00:00.000Z",
    "entity": "user",
    "id": "4",
    "modified": "2020-06-29T00:00:00.000Z",
    "name": "mock user name",
    "privateDrawingCount": 0,
    "publicDrawingCount": 0,
  },
]
`;

exports[`bulkGetUser 2`] = `
Object {
  "Count": 5,
  "Items": Array [
    Object {
      "_ct": "2020-06-29T00:00:00.000Z",
      "_et": "user",
      "_md": "2020-06-29T00:00:00.000Z",
      "dc": 0,
      "n": "mock user name",
      "pdc": 0,
      "pk": "user#3",
      "sk": "user#3",
    },
    Object {
      "_ct": "2020-06-29T00:00:00.000Z",
      "_et": "user",
      "_md": "2020-06-29T00:00:00.000Z",
      "dc": 0,
      "n": "mock user name",
      "pdc": 0,
      "pk": "user#4",
      "sk": "user#4",
    },
    Object {
      "_ct": "2020-06-29T00:00:00.000Z",
      "_et": "user",
      "_md": "2020-06-29T00:00:00.000Z",
      "dc": 0,
      "n": "mock user name",
      "pdc": 0,
      "pk": "user#2",
      "sk": "user#2",
    },
    Object {
      "_ct": "2020-06-29T00:00:00.000Z",
      "_et": "user",
      "_md": "2020-06-29T00:00:00.000Z",
      "dc": 0,
      "n": "mock user name",
      "pdc": 0,
      "pk": "user#0",
      "sk": "user#0",
    },
    Object {
      "_ct": "2020-06-29T00:00:00.000Z",
      "_et": "user",
      "_md": "2020-06-29T00:00:00.000Z",
      "dc": 0,
      "n": "mock user name",
      "pdc": 0,
      "pk": "user#1",
      "sk": "user#1",
    },
  ],
  "ScannedCount": 5,
}
`;
